woocommerce/woocommerce
# registerProductBlock: Always register the block

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This fixes an issue whereby if you reloaded the page/URL on the templates data view, it would not correctly register product blocks.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Verify blocks using `registerProductBlockType` registers correctly in both single product templates and post/page editor
2. Check transitions between templates using both the command UI (CMD+K) and manual navigation maintain correct block behavior
3. Reload the templates data-view to ensure template previews register blocks correctly
4. Verify existing block implementations continue to work without changes
5. Ensure E2E tests pass



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
